rust/worker/src/execution/operators/write_segments.rs

```diff
@@ -59,9 +62,9 @@ impl WriteSegmentsOperator {
 
 #[derive(Debug)]
 pub struct WriteSegmentsInput {
-    record_segment_writer: RecordSegmentWriter,
-    hnsw_segment_writer: Box<DistributedHNSWSegmentWriter>,
-    metadata_segment_writer: MetadataSegmentWriter<'static>,
+    record_segment_writer: Option<RecordSegmentWriter>,
```

Comment:
Currently there is no need for individually wrapped writers. I'm using `None` to represent the scenario where no records have been added to the collection (and hence no collection dimension and hnsw writer), and in this scenario no writer is needed. So far I have not realized a scenario where individually wrapped writers could help.

Reply:
Right, so if there is no need for individually wrapping. Maybe it makes sense to make a nested struct that holds all the writers and make that an option. So that we only model the valid states